scripts-endpoint.md

```diff
@@ -0,0 +1,169 @@
+  "_embedded" : {
+    "scripts" [
+      {
+        "name" : "import",
+        "description" : "Import items into DSpace"
+        "_links" : {
+          "self" : {
+            "href" : "/api/admin/scripts/import"
+          }
+        }
+      },
+      {
+        "name" : "metadata-import",
+        "description" : "Import metadata after batch editing"
+        "_links" : {
+          "self" : {
+            "href" : "/api/admin/scripts/metadata-import"
+          }
+        }
+      },
+      ...
+    ]
+  }
+```
+
+## Script Name Endpoint
+
+### Script Details
+**GET /api/admin/scripts/<:script-name>**
+
+This endpoint will return information on all the parameters that are required to invoke the script. This dependens on the parameters defined in the implementation of the script.
+
+The JSON response document is as follows
+```
+{
+   "name" : "import",
+   "description" : "Import items into DSpace",
+   "parameters" : [
+    {
+      "name" : "a",
+      "description" : "add items to DSpace"
+      "type" : "boolean"
+    },
+    {
+      "name" : "b",
+      "description" : "add items to DSpace via Biblio-Transformation-Engine (BTE)"
+      "type" : "boolean"
+    },
+    {
+      "name" : "c",
+      "description" : "destination collection(s) database ID"
+      "type" : "id"
+    },
+    {
+      "name" : "d",
+      "description" : "delete items listed in mapfile"
+      "type" : "file"
+    },
+    {
+      "name" : "e",
+      "description" : "email of eperson doing importing"
+      "type" : "string"
+    },
+    {
+      "name" : "i",
+      "description" : "input type in case of BTE import"
+      "type" : "string"
+    },
+    {
+      "name" : "n",
+      "description" : "if sending submissions through the workflow, send notification emails"
+      "type" : "boolean"
+    },
+    {
+      "name" : "p",
+      "description" : "apply template"
+      "type" : "string"
+    },
+    {
+      "name" : "q",
+      "description" : "don't display metadata"
+      "type" : "boolean"
+    },
+    {
+      "name" : "m",
+      "description" : "mapfile items in mapfile"
+      "type" : "file"
+    },
+    {
+      "name" : "r",
+      "description" : "replace items in mapfile"
+      "type" : "boolean"
+    },
+    {
+      "name" : "R",
+      "description" : "resume a failed import (add only)"
+      "type" : "boolean"
+    },
+    {
+      "name" : "t",
+      "description" : "test run - do not actually import items"
+      "type" : "boolean"
+    },
+    {
+      "name" : "w",
+      "description" : "send submission through collection's workflow"
+      "type" : "boolean"
+    },
+    {
+      "name" : "z",
+      "description" : "zip file containig the import"
+      "type" : "file"
+    }
+   ]
+}
+```
+
+Following parameter types are available:
+* `string`: A regular string value
+* `date`: A string that has a valid date pattern
+* `boolean`: true or false
+* `file`: For parameters with this type, the user has to provide a file
```

Comment:
we should specify how file as supplied. Obviously one option is to provide them in a multipart POST request using the parameter as name of the file but we want to document that

Reply:
I'll document the multipart post instructions. The ultimate goal for me would be to have file upload support using something like the [tus protocol](https://tus.io/demo.html). Because then people can upload for example really large imports and datasets. But that's probably something for DSpace 7.1 or DSpace 8.

Reply:
I will like to have support for resumable upload but it is probably out-of-scope for dspace 7. Once introduced it should be added to all the upload endpoints (including the submission)